OpenEdXMobile/src/main/java/org/edx/mobile/social/SocialLoginDelegate.java

```diff
@@ -199,6 +211,11 @@ public ProfileModel call() throws Exception {
                 try {
                     auth = loginAPI.logInUsingGoogle(accessToken);
                 } catch (LoginAPI.AccountNotLinkedException e) {
+                    if (feature == Feature.SIGIN_IN && e.getResponseCode() ==  HttpURLConnection.HTTP_BAD_REQUEST) {
+                        final String title =  activity.getResources().getString(R.string.login_error);
+                        final CharSequence desc = ResourceUtil.getFormattedString(context.getResources(), R.string.error_account_not_linked_desc_google_2, descParams);
+                        throw new LoginException(new LoginErrorMessage(title, desc.toString()));
```

Comment:
This code block is being repeated with minimal string changes, so, it could be made into a function with different arguments and used instead of copying the boilerplate code.